File: backend-dotnet/AspNetBackend/Entities/Player.cs
using System.ComponentModel.DataAnnotations;

namespace AspNetBackend.Entities;

public class Player
{
    public long Id { get; set; }

    [MaxLength(50)]
    public string? Name { get; set; } = string.Empty;

    [StringLength(200)]
    public string? PasswordHash { get; set; } = string.Empty;

    [StringLength(200)]
    public string? RefreshToken { get; set; }
    public DateTime? RefreshTokenExpiryTime { get; set; }

    [EmailAddress]
    [MaxLength(100)]
    public string? UserEmail { get; set; }

    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    
}

File: backend-java/src/main/java/com/example/backendjava/model/Player.java
package com.example.backendjava.model;


import jakarta.persistence.*;

import java.time.LocalDateTime;

@Entity
@Table(name = "players", schema = "players")
@org.hibernate.annotations.Immutable
public class Player {
    @Id
    private Long id;

    @Column(name = "name")
    private String name;

    @Column(name = "password_hash")
    private String passwordHash;

    @Column(name = "refresh_token")
    private String refreshToken;

    @Column(name = "refresh_token_expiry_time")
    private LocalDateTime refreshTokenExpiryTime;

    @Column(name = "user_email")
    private String userEmail;

    @Column(name = "created_at")
    private LocalDateTime createdAt;

    public Player() {}

    public Player(String name) {
        this.name = name;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }
}
